src/project/builtin_plugins.rs

```diff
@@ -0,0 +1,97 @@
+use cairo_lang_defs::plugin::{MacroPlugin, MacroPluginMetadata, PluginResult};
+use cairo_lang_executable::plugin::executable_plugin_suite;
+use cairo_lang_semantic::plugin::PluginSuite;
+use cairo_lang_starknet::starknet_plugin_suite;
+use cairo_lang_syntax::node::ast::ModuleItem;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_test_plugin::{test_assert_suite, test_plugin_suite};
+use scarb_metadata::{CompilationUnitCairoPluginMetadata, Metadata};
+
+/// Representation of known built-in plugins available in the Cairo compiler.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum BuiltinPlugin {
+    AssertMacros,
+    Executable,
+    CairoRun,
+    CairoTest,
+    Starknet,
+}
+
+impl BuiltinPlugin {
+    /// Creates a new instance of `BuiltinPlugin` corresponding to the given
+    /// [`CompilationUnitCairoPluginMetadata`].
+    /// Returns `None` if `plugin_metadata` does not describe any known built-in plugin.
+    pub fn from_plugin_metadata(
+        metadata: &Metadata,
+        plugin_metadata: &CompilationUnitCairoPluginMetadata,
+    ) -> Option<Self> {
+        // Discard those plugins which are not built-in
+        // before checking their discriminators in the next step.
+        if !metadata.is_builtin(plugin_metadata).unwrap_or_default() {
+            return None;
+        }
+
+        // The package discriminator has form: "<name> <version> (<source>)".
+        let package_discriminator = &plugin_metadata.package.repr;
+
+        // Check if the discriminator refers to a known built-in plugin package.
+        if package_discriminator.contains("assert_macros") {
```

Comment:
```suggestion
        if package_discriminator.starts_with("assert_macros ") {
```

this will be immune to packages having this keyword in their source part

Reply:
Done, a bit differently, but with the same result